network/src/main/java/HandleMailToClick.java

```diff
@@ -0,0 +1,87 @@
+/*
+ *  Copyright 2020, TeamDev. All rights reserved.
+ *
+ *  Redistribution and use in source and/or binary forms, with or without
+ *  modification, must retain the above copyright notice and the following
+ *  disclaimer.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ *  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ *  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ *  A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ *  OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ *  SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ *  LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ *  DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ *  THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ *  (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ *  OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+import com.teamdev.jxbrowser.browser.Browser;
+import com.teamdev.jxbrowser.engine.Engine;
+import com.teamdev.jxbrowser.engine.EngineOptions;
+import com.teamdev.jxbrowser.net.callback.BeforeUrlRequestCallback;
+import com.teamdev.jxbrowser.net.callback.BeforeUrlRequestCallback.Response;
+import com.teamdev.jxbrowser.view.swing.BrowserView;
+
+import javax.swing.*;
+import java.awt.*;
+import java.awt.event.WindowAdapter;
+import java.awt.event.WindowEvent;
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+
+import static com.teamdev.jxbrowser.engine.RenderingMode.HARDWARE_ACCELERATED;
+import static java.awt.Desktop.isDesktopSupported;
+
+/**
+ * This example demonstrates how to handle the clicks on the {@code mailto:} links to open
+ * a default desktop mail client.
+ */
+public final class HandleMailToClick {
+
+    private static final String HTML = "<a href=\"mailto:test@example.com\">Click to mail</a>";
+
+    public static void main(String[] args) {
+        Engine engine = Engine.newInstance(
+                EngineOptions.newBuilder(HARDWARE_ACCELERATED).build());
+        Browser browser = engine.newBrowser();
+
+        engine.network().set(BeforeUrlRequestCallback.class, params -> {
+            String url = params.urlRequest().url();
+            if (url.startsWith("mailto:")) {
+                Desktop desktop = Desktop.getDesktop();
+                if (isDesktopSupported() && desktop.isSupported(Desktop.Action.MAIL)) {
```

Comment:
Consider checking `isDesktopSupported` before calling `Desktop.getDesktop()`. In other case, you may get an `UnsupportedOperationException`:
https://docs.oracle.com/javase/9/docs/api/java/awt/Desktop.html#getDesktop--